pkg/apis/eksctl.io/v1alpha5/validation_test.go

```diff
@@ -46,9 +46,98 @@ var _ = Describe("ConfigFile ssh flags validation", func() {
 		err := validateNodeGroupSSH(nil)
 		Expect(err).ToNot(HaveOccurred())
 	})
+
+	Context("Instances distribution", func() {
+
+		var ng *NodeGroup
+		BeforeEach(func() {
+			ng = &NodeGroup{
+				InstancesDistribution: &NodeGroupInstancesDistribution{
+					InstanceTypes: []string{"t3.medium", "t3.large"},
+				},
+			}
+		})
+
+		It("It doesn't panic when instance distribution is not enabled", func() {
+			ng.InstancesDistribution = nil
+			err := validateInstancesDistribution(ng)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("It fails when instance distribution is enabled and instanceType is not empty or \"mixed\"", func() {
+			err := validateInstancesDistribution(ng)
+			Expect(err).ToNot(HaveOccurred())
+
+			ng.InstanceType = "t3.small"
+
+			err = validateInstancesDistribution(ng)
+			Expect(err).To(HaveOccurred())
+		})
+
+		It("It fails when the instance distribution doesn't have at least 2 different instance types", func() {
+			ng.InstanceType = "mixed"
+			ng.InstancesDistribution.InstanceTypes = []string{"t3.medium", "t3.medium"}
+
+			err := validateInstancesDistribution(ng)
+			Expect(err).To(HaveOccurred())
+
+			ng.InstanceType = "mixed"
+			ng.InstancesDistribution.InstanceTypes = []string{"t3.medium", "t3.small"}
+
+			err = validateInstancesDistribution(ng)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("It fails when the onDemandBaseCapacity is not above 0", func() {
```

Comment:
Really cosmetic one. Test description is a bit misleading. Code checks actually for <0. 0 is a supported value for onDemandBaseCapacity.

Reply:
Thanks, I'll fix it.